src/ripple/core/impl/SQLInterface_shards.cpp

```diff
@@ -0,0 +1,1383 @@
+        lgr->getSession() << boost::str(
+            boost::format("PRAGMA cache_size=-%d;") %
+            kilobytes(config.getValueFor(SizedItem::lgrDBCache, boost::none)));
+
+        return {true, std::move(tx), std::move(lgr)};
+    }
+    else
+    {
+        // transaction database
+        auto tx{std::make_unique<DatabaseCon>(
+            setup, TxDBName, TxDBPragma, TxDBInit, checkpointerSetup)};
+        tx->getSession() << boost::str(
+            boost::format("PRAGMA cache_size=-%d;") %
+            kilobytes(config.getValueFor(SizedItem::txnDBCache)));
+
+        // ledger database
+        auto lgr{std::make_unique<DatabaseCon>(
+            setup, LgrDBName, LgrDBPragma, LgrDBInit, checkpointerSetup)};
+        lgr->getSession() << boost::str(
+            boost::format("PRAGMA cache_size=-%d;") %
+            kilobytes(config.getValueFor(SizedItem::lgrDBCache)));
+
+        if (setupFromConfig &&
+            (!setup.standAlone || setup.startUp == Config::LOAD ||
+             setup.startUp == Config::LOAD_FILE ||
+             setup.startUp == Config::REPLAY))
+        {
+            // Check if AccountTransactions has primary key
+            std::string cid, name, type;
+            std::size_t notnull, dflt_value, pk;
+            soci::indicator ind;
+            soci::statement st =
+                (tx->getSession().prepare
+                     << ("PRAGMA table_info(AccountTransactions);"),
+                 soci::into(cid),
+                 soci::into(name),
+                 soci::into(type),
+                 soci::into(notnull),
+                 soci::into(dflt_value, ind),
+                 soci::into(pk));
+
+            st.execute();
+            while (st.fetch())
+            {
+                if (pk == 1)
+                {
+                    return {false, std::move(tx), std::move(lgr)};
+                }
+            }
+        }
+
+        return {true, std::move(tx), std::move(lgr)};
+    }
+}
+
+boost::optional<LedgerIndex>
+getMinLedgerSeq(soci::session& db, SQLInterface::TableType type)
+{
+    std::string query =
+        "SELECT MIN(LedgerSeq) FROM " + SQLInterface::tableName(type) + ";";
+    boost::optional<LedgerIndex> m;
+    db << query, soci::into(m);
+    return m;
+}
+
+boost::optional<LedgerIndex>
+getMaxLedgerSeq(soci::session& db, SQLInterface::TableType type)
+{
+    std::string query =
+        "SELECT MAX(LedgerSeq) FROM " + SQLInterface::tableName(type) + ";";
+    boost::optional<LedgerIndex> m;
+    db << query, soci::into(m);
+    return m;
+}
+
+void
+deleteByLedgerSeq(
+    soci::session& db,
+    SQLInterface::TableType type,
+    LedgerIndex ledgerSeq)
+{
+    db << "DELETE FROM " << SQLInterface::tableName(type)
+       << " WHERE LedgerSeq == " << ledgerSeq << ";";
+}
+
+void
+deleteBeforeLedgerSeq(
+    soci::session& db,
+    SQLInterface::TableType type,
+    LedgerIndex ledgerSeq)
+{
+    std::string query = "DELETE FROM " + SQLInterface::tableName(type) +
```

Comment:
This is mixing string concatenation and boost format. Why not just skip the boost format part and add the ledgerSeq when creating the string?

Reply:
Done.